### PR TITLE
First drop of mpReactiveMessaging-3.0

### LINKED ISSUE #25386: Create mpReactiveMessaging-3.0 feature
- Create mpReactiveMessaging-3.0 feature as noship
- Pull in new version of SmallRye Reactive Messaging
- Update unit tests
- make sure it all builds

### PR DESCRIPTION
**Pull request by tevans78**
This PR creates the mpReactiveMessaging-3.0 feature, backed by SmallRye Reactive Messaging Provider 4.8.0.

Only `BasicReactiveMessagingTest` is repeated to test this. Others are still failing and will need follow up work.

fixes #25386

### CODE REVIEW

**Comment by posmith** [comment]
L2 message review not required

**Comment by tevans78** [comment]
#build

File: dev/io.openliberty.io.smallrye.reactive.messaging-provider4/src/io/openliberty/microprofile/reactive/messaging30/internal/OLReactiveMessaging30Extension.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import io.smallrye.reactive.messaging.providers.impl.ConnectorFactories;
+import io.smallrye.reactive.messaging.providers.impl.InternalChannelRegistry;
+import io.smallrye.reactive.messaging.providers.wiring.Wiring;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.AnnotatedType;
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
+import jakarta.enterprise.inject.spi.Extension;
+
+@Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM", "application.bdas.visible=true" })
```
Comment (hutchig) [inline dev/io.openliberty.io.smallrye.reactive.messaging-provider4/src/io/openliberty/microprofile/reactive/messaging30/internal/OLReactiveMessaging30Extension.java:42]: https://ibm-cloud.slack.com/archives/C31DW78RG/p1633010438267800
Thomas Watson:
I've been touching various service components in Open Liberty and often times I convert a bnd.bnd instruction for Service-Component to use annotations instead.  Most all the time I do that I leave the service property service.vendor=IBM out of the annotation for @Component.  I see no value to this service property.  It certainly seems strange at least on an open source project.  Is there any reason we need to keep this property on all our service components?




2 replies


Alasdair Nottingham
  [2 years ago](https://ibm-cloud.slack.com/archives/C31DW78RG/p1633010843267900?thread_ts=1633010438.267800&cid=C31DW78RG)
I don't think we need to keep it


Thomas Watson
  [2 years ago](https://ibm-cloud.slack.com/archives/C31DW78RG/p1633012652268100?thread_ts=1633010438.267800&cid=C31DW78RG)
ok good

Comment (tevans78) [reply dev/io.openliberty.io.smallrye.reactive.messaging-provider4/src/io/openliberty/microprofile/reactive/messaging30/internal/OLReactiveMessaging30Extension.java:42]: OK, fair point ... I might do a separate PR to remove it from here and elsewhere.

File: dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java [hunk 2]
```diff
@@ -130,6 +130,8 @@ private static Set<String> getCompatibleFeatures(Set<String> allFeatures, boolea
         compatibleFeatures.remove("springBoot-2.0");
         compatibleFeatures.remove("springBoot-3.0"); // springBoot 3.0 only supports EE 10
 
+        compatibleFeatures.remove("mpReactiveMessaging-3.0"); //still in development
+
```
Comment (Azquelt) [inline dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java:134]: Why isn't it compatible yet?

**Review by Azquelt — APPROVED** [review]
I'd like to know why the feature has to be excluded from `EE9FeatureCompatibilityTest` but otherwise this looks good.

File: dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java (hunk 2 above)
Comment (tevans78) [reply dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java:134]: A lack of testing at the time when this PR was first created. Even now, I have only repeated one test on that.

**Pull request merged**